File: server/router/cloudpods/k8s/deployment.go
package k8s

import (
	v1 "ygpt/server/api/v1"

	"github.com/gin-gonic/gin"
)

type K8sDeploymentRouter struct {
}

func (*K8sDeploymentRouter) InitK8sDeploymentRouter(Router *gin.RouterGroup) {
	deploymentGroup := Router.Group("cloud/k8s/deployment")
	deploymentApi := v1.ApiGroupApp.CloudApiGroup.K8sAPiGroup.DeploymentAPI
	{
		deploymentGroup.POST("list", deploymentApi.DeploymentList)                 // 获取deployment列表
		deploymentGroup.GET("webConsole", deploymentApi.GetWebConsoleSSH)          // 远程终端访问
		deploymentGroup.POST("create", deploymentApi.DeploymentCreate)             // 创建deployment
		deploymentGroup.DELETE("delete", deploymentApi.DeploymentDelete)           // 删除deployment
		deploymentGroup.POST("sshInfo", deploymentApi.GetDeploymentSshInfo)        //获取容器的ssh信息
		deploymentGroup.GET("pods", deploymentApi.GetDeploymentPods)               //
		deploymentGroup.POST("getK8sNodeList", deploymentApi.GetK8sNodeList)       //获取集群节点列表
		deploymentGroup.POST("modK8sNodeStatus", deploymentApi.ModK8sNodeStatus)   //修改集群相关节点调度状态
		deploymentGroup.POST("deploymentMonitor", deploymentApi.DeploymentMonitor) //监控容器
	}
}

File: web/src/api/cloudpods/k8s/deployment.js
//* Deployment（无状态） 相关API
import request from "@/utils/request"

/**
 * @description 获取Deployment列表
 * @param {*} params 请求参数
 * @returns AjaxResult
 */
export function listDeployment(params) {
    return request({
        url: 'cloud/k8s/deployment/list',
        method: 'post',
        data: params
    })
}

/**
 * @description 添加Deployment
 * @param {*} data 表单数据
 * @returns AjaxResult
 */
export function addDeployment(data) {
    return request({
        url: 'cloud/k8s/deployment/create',
        method: 'post',
        data: data
    })
}

/**
 * @description 删除Deployment
 * @param {*} data
 * @returns
 */
export function deleteDeployment(data) {
    return request({
        url: 'cloud/k8s/deployment/delete',
        method: 'delete',
        data: data
    })
}

/**
 * @description 获取WebTerminal连接信息
 * @param {int} params
 */
export function webConnectDeployment(params) {
    return request({
        url: 'cloud/k8s/deployment/webConsole',
        method: 'get',
        params: params
    })
}

export function getDeploymentPods(params){
    return request({
        url: 'cloud/k8s/deployment/pods',
        method: 'get',
        params: params
    })
}
//获取集群节点列表
export function getK8sNodeList(params){
    return request({
        url: 'cloud/k8s/deployment/getK8sNodeList',
        method: 'post',
        data: params
    })
}
//修改集群相关节点调度状态
export function modK8sNodeStatus(params){
    return request({
        url: 'cloud/k8s/deployment/modK8sNodeStatus',
        method: 'post',
        data: params
    })
}
//监控集群容器
export function deploymentMonitor(params){
    return request({
        url: 'cloud/k8s/deployment/deploymentMonitor',
        method: 'post',
        data: params
    })
}